packages/sdk-client-v3/src/utils/executor.ts

```diff
@@ -68,17 +70,31 @@ export default async function executor(request: HttpClientConfig) {
       }
 
       async function executeWithRetry<T = any>(): Promise<T> {
-        // first attempt
-        let _response = await execute()
-
-        if (predicate(retryCodes, _response)) return _response
+        const executeWithTryCatch = async (retryCodes) => {
+          let _response = {} as any
+          try {
+            _response = await execute()
+            if (predicate(retryCodes, _response)) return _response
+          } catch (e) {
+            if (e.name.includes('AbortError')) {
+              _response = e
+            } else {
+              throw e
+            }
+          }
+          return _response
+        }
 
+        // first attempt
+        let _response = await executeWithTryCatch(retryCodes)
         // retry attempts
-        while (enableRetry && retryCount < maxRetries) {
+        while (
+          enableRetry &&
+          (retryOnAbort || !abortController || !abortController.signal) &&
```

Comment:
What happens when `retryOnAbort` is truthy and `aborController` is falsy and vice versa?

Reply:
Not really, I don't think these two are the same, considering where the both of them are situated. One the v2 is to retry a network error, in the finally error callback, inside that callback we are certain that it's a network error whilst in the v3 the check is done, in the executor (whose error might not specifically be a network error).

Also, the slightly different logic check here is telling, I can decide to have a retry for a 503 error (that is enableRetry will be set to true) but at the same time I don't want to enable any retryOnAbort (retryOnAbort will be false) at this combination, the customer won't get any retry. Is this correct?

I think we need to have a very good look at this check and all the possible truthy/falsey case for each combination of the individual variables like the `enableRetry`, `retryOnAbort` `abortController` etc, so we are sure the executor function behaves as intended.

Reply:
https://github.com/commercetools/commercetools-sdk-typescript/pull/714/commits/f6b22c85647537d07299bea84be926ab116edacb